src/components/ui/Form.tsx

```diff
@@ -0,0 +1,153 @@
+  TName extends FieldPath<TFieldValues> = FieldPath<TFieldValues>,
+> = {
+  name: TName;
+};
+
+const FormFieldContext = createContext<FormFieldContextValue>({} as FormFieldContextValue);
+
+const FormField = <
+  TFieldValues extends FieldValues = FieldValues,
+  TName extends FieldPath<TFieldValues> = FieldPath<TFieldValues>,
+>({
+  ...props
+}: ControllerProps<TFieldValues, TName>) => {
+  return (
+    <FormFieldContext.Provider value={{ name: props.name }}>
+      <Controller {...props} />
+    </FormFieldContext.Provider>
+  );
+};
+
+const useFormField = () => {
+  const fieldContext = useContext(FormFieldContext);
+  const itemContext = useContext(FormItemContext);
+  const { getFieldState, formState } = useFormContext();
+
+  const fieldState = getFieldState(fieldContext.name, formState);
+
+  if (!fieldContext) {
+    throw new Error('useFormField should be used within <FormField>');
+  }
+
+  const { id } = itemContext;
+
+  return {
+    id,
+    name: fieldContext.name,
+    formItemId: `${id}-form-item`,
+    formMessageId: `${id}-form-item-message`,
+    ...fieldState,
+  };
+};
+
+type FormItemContextValue = {
+  id: string;
+};
+
+const FormItemContext = createContext<FormItemContextValue>({} as FormItemContextValue);
+
+const FormItem = forwardRef<HTMLDivElement, HTMLAttributes<HTMLDivElement>>(
+  ({ className, ...props }, ref) => {
+    const id = useId();
+
+    return (
+      <FormItemContext.Provider value={{ id }}>
+        <div ref={ref} className={className} {...props} />
+      </FormItemContext.Provider>
+    );
+  },
+);
+FormItem.displayName = 'FormItem';
+
+const FormLabel = forwardRef<
+  ElementRef<typeof LabelPrimitiveRoot>,
+  ComponentPropsWithoutRef<typeof LabelPrimitiveRoot> & {
+    required?: boolean;
+  }
+>(({ className, children, required, ...props }, ref) => {
+  const { formItemId } = useFormField();
+
+  return (
+    <Label ref={ref} className={cn('mb-2', className)} htmlFor={formItemId} {...props}>
+      {children}
+      {required && <span className="text-danger">*</span>}
+    </Label>
+  );
+});
+FormLabel.displayName = 'FormLabel';
+
+const FormControl = forwardRef<ElementRef<typeof Slot>, ComponentPropsWithoutRef<typeof Slot>>(
+  ({ ...props }, ref) => {
+    const { error, formItemId } = useFormField();
+
+    return (
+      <Slot
+        ref={ref}
+        id={formItemId}
+        // aria-describedby={!error ? `${formDescriptionId}` : `${formDescriptionId} ${formMessageId}`}
+        aria-invalid={!!error}
+        {...props}
+      />
+    );
+  },
+);
+FormControl.displayName = 'FormControl';
+
+const FormMessage = forwardRef<HTMLParagraphElement, HTMLAttributes<HTMLParagraphElement>>(
+  ({ className, children, ...props }, ref) => {
+    const { error, formMessageId } = useFormField();
+    const body = error ? String(error?.message) : children;
+
+    if (!body) {
+      return null;
+    }
+
+    return (
+      <div className="mt-1 flex items-start gap-1.5">
```

Comment:
I just wondered, could you kindly share with me the reason why you didn't use `items-center` to align both icon and message, and instead used `items-start`?
With using `items-center` to both icon and message, i guess it'd be like this↓
```tsx
<div className="mt-1 flex items-center gap-1.5">
        <Icon icon={ExclamationIcon} size={3.5} color="danger" />
        <p
          ref={ref}
          id={formMessageId}
          className={cn('text-danger leading-5', className)}
          {...props}
        >
          {body}
        </p>
      </div>
```
↓Original implementation
<img width="169" alt="Screenshot 2024-01-13 at 0 14 20" src="https://github.com/genesis-tech-tribe/nishiki-frontend/assets/99339182/2b5b036d-e5e9-4cd9-9d4d-c48793a5261d">

↓ With `items-center`
<img width="168" alt="Screenshot 2024-01-13 at 0 16 00" src="https://github.com/genesis-tech-tribe/nishiki-frontend/assets/99339182/d4d75cb8-d02a-4413-8b70-3f83225e9f35">

Reply:
Good question! It's because when the message spans more than two lines, the alignment becomes misaligned. So, `items-start` is necessary. That is why the `line-height` of the message and the height of the icon area have the same height.

Reply:
Makes sense ! Thank you for detailed explanation!